### PR TITLE
[TagInput] leftIconName + rightElement

### PR DESCRIPTION
**Pull request by giladgray**
#### Changes proposed in this pull request:

- `leftIconName` renders an icon as first child
- `rightElement` appears as last child
- just like `InputGroup`

screen shot 2017-08-07 at 8 47 27 pm

### CODE REVIEW

**Comment by cmslewis** [comment]
@giladgray can we also add support for `pt-round` in this PR, while we're on the topic of `TagInput` aesthetic extensions? That would have been the final necessary style piece for our side project last week.

**Comment by cmslewis** [comment]
FYI Clear button doesn't look centered in the button in your screenshot: 
image

File: packages/labs/src/components/tag-input/tagInput.tsx [hunk 1]
```diff
@@ -149,10 +159,24 @@ export class TagInput extends AbstractComponent<ITagInputProps, ITagInputState>
...
     }
 
+    private maybeRenderLeftIcon(useLarge: boolean) {
+        const { leftIconName } = this.props;
+        if (leftIconName == null) {
+            return undefined;
+        }
+        const iconClass = classNames(
+            Classes.TAG_INPUT_ICON,
+            useLarge ? CoreClasses.ICON_LARGE : CoreClasses.ICON_STANDARD ,
```
Comment (cmslewis) [inline packages/labs/src/components/tag-input/tagInput.tsx]: nit: no space before comma

**Comment by giladgray** [comment]
addressed all the issues: button centering and padding and whitespace.

@cmslewis i poked around `.pt-round` support and it's not so easy. i think we should instead try to merge the `.pt-input-group` and `.pt-tag-input` styles and ensure that they both support the same things, _forever_.

File: packages/labs/src/components/tag-input/_tag-input.scss [hunk 2]
```diff
@@ -9,39 +9,56 @@
 @import "~@blueprintjs/core/src/components/forms/common";
 @import "~@blueprintjs/core/src/components/tag/common";
 
-$tag-input-padding: ($pt-input-height - $tag-height) / 2;
+$ti-padding: ($pt-input-height - $tag-height) / 2;
+$ti-right-padding: ($pt-input-height - $pt-button-height-small) / 2;
```
Comment (llorca) [inline packages/labs/src/components/tag-input/_tag-input.scss]: So much nicer with this added 👏 Perfectly squared off. `$tag-input-` ain't that long is it?

How about `-right-padding` -> `-padding-right` to match the property name?

**Comment by giladgray** [comment]
👀 on https://github.com/palantir/blueprint/pull/1420/commits/e042aa84c4f73fa7242aafcd2e95278d40c70621 would be appreciated, to make sure nothing is weird

**Pull request merged**